src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -43,18 +43,21 @@ public virtual bool TryGetPrimaryKey(out string key)
         private IEnumerable<NamedValue> GetFields()
         {
             foreach (var fieldName in Type.FieldNames)
-            {
-                var formulaValue = GetField(fieldName);
-                yield return new NamedValue(fieldName, formulaValue);
+            {
+                // below will always succeed.

```

Comment:
>// below will always succeed [](http://example.com/codeflow?start=16&length=28)

How can we guarantee that this will always succeed? AggregateType.TryGetBackingDType can return false.

Reply:
`TryGetBackingDType` can return false but since fieldName is being enumerated from `Type` TryGetBackingDType should always succeed here.